contracts/Market.sol

```diff
@@ -70,7 +74,12 @@ contract Market is IMarket, Ownable, Pausable {
 
     // The beneficiary can retrieve already released funds via this.
     function withdrawAvailableFunds() external payable {
-        _beneficiary.transfer(this.balance);
+        if (totalWithdrawn >= beneficiaryLimit)
+            owner().transfer(this.balance);
+        else {
+            totalWithdrawn += this.balance;
```

Comment:
This allows the beneficiary to withdraw the total amount of `this.balance` regardless of how big it is. We need to cap the withdrawal amount at `beneficiaryLimit`, or at whatever portion of that limit hasn't been withdrawn yet.

Reply:
`this.balance` is reset after transfer.

Reply:
Fixed.